pkg/devfile/generator/utils.go

```diff
@@ -168,13 +169,11 @@ func addSyncRootFolder(container *corev1.Container, sourceMapping string) string
 func addSyncFolder(container *corev1.Container, sourceVolumePath string, projects []v1.Project) error {
 	var syncFolder string
 
-	// if there are no projects in the devfile, source would be synced to $PROJECTS_ROOT
 	if len(projects) == 0 {
-		syncFolder = sourceVolumePath
+		// No projects found, set PROJECT_SOURCE to empty
+		syncFolder = ""
```

Comment:
I don't believe we need any changes here, the only change that is required is the `container.Env` change you made below to insert the special env at the start of the slice. 

In addition I think it may be best to revert and include all the deleted comments, reviewing this issue was slightly confusing so it may be good to over document this portion.